Lib/test/test_textwrap.py

```diff
@@ -1076,5 +1077,49 @@ def test_first_word_too_long_but_placeholder_fits(self):
         self.check_shorten("Helloo", 5, "[...]")
 
 
+class WideCharacterTestCase(BaseTestCase):
+    def text_len(self, text):
+        n = 0
+        for c in text:
+            if unicodedata.east_asian_width(c) in {'F', 'W'}:
+                n += 2
+            else:
+                n += 1
+        return n
```

Comment:
I know this is just an example function for a test, but it’s also the main use case from the other ticket, and people may copy it, so could we have a more efficient version?

```python
return sum(
    2 if unicodedata.east_asian_width(c) in {'F', 'W'} else 1
    for c in text
)
```